<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="state_topic_name", default="/state"/>
    <arg name="llm_model_name" default="Qwen/Qwen3-32B"/>
    <arg name="event_queue_max_tokens" default="8000"/>
    <arg name="continuous_queue_max_tokens" default="2000"/>
    <arg name="event_topics" default="['/asr', '/user_commands', '/navigation_goals']"/>
    <arg name="continuous_topics" default="['/vision', '/sensors', '/pose']"/>
    <arg name="long_term_memory_file_pth" default=""/>
    <arg name="state_file_pth" default=""/>

    <node pkg="state_manager" exec="state_manager" name="state_manager">
        <param name="state_topic_name" value="$(var state_topic_name)"/>
        <param name="llm_model_name" value="$(var llm_model_name)"/>
        <param name="event_queue_max_tokens" value="$(var event_queue_max_tokens)"/>
        <param name="continuous_queue_max_tokens" value="$(var continuous_queue_max_tokens)"/>
        <param name="event_topics" value="$(var event_topics)"/>
        <param name="continuous_topics" value="$(var continuous_topics)"/>
        <param name="long_term_memory_file_pth" value="$(var long_term_memory_file_pth)"/>
        <param name="state_file_pth" value="$(var state_file_pth)"/>
    </node>
</launch>